File: server/makc2022--nodejs--nestjs/src/layer4/domains/dummy-main/queries/item/get/domain-item-get-query-output.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { DummyMainDummyManyToManyEntityObject } from '../../../../../../layer3/nosql-mongo/sample/entities/dummy-main-dummy-many-to-many/dummy-main-dummy-many-to-many-entity.object';
import { DummyMainEntityObject } from '../../../../../../layer3/nosql-mongo/sample/entities/dummy-main/dummy-main-entity.object';
import { DummyManyToManyEntityObject } from '../../../../../../layer3/nosql-mongo/sample/entities/dummy-many-to-many/dummy-many-to-many-entity.object';
import { DummyOneToManyEntityObject } from '../../../../../../layer3/nosql-mongo/sample/entities/dummy-one-to-many/dummy-one-to-many-entity.object';

/** Выходные данные запроса на получение элемента в домене. */
export class DomainItemGetQueryOutput {
  /** Объект сущности "DummyMain". */
  objectOfDummyMainEntity: DummyMainEntityObject;

  /** Объекты сущности "DummyManyToMany". */
  objectsOfDummyManyToManyEntity: DummyManyToManyEntityObject[];

  /** Объекты сущности "DummyMainDummyManyToMany". */
  objectsOfDummyMainDummyManyToManyEntity: DummyMainDummyManyToManyEntityObject[];

  /** Объект сущности "DummyOneToMany". */
  objectOfDummyOneToManyEntity: DummyOneToManyEntityObject;
}

File: server/makc2022--dotnet/Makc2022.Layer4.Sql.Domains.DummyMain/Queries/Item/Get/DomainItemGetQueryOutput.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer3.Sql.Sample.Entities.DummyMain;
using Makc2022.Layer3.Sql.Sample.Entities.DummyMainDummyManyToMany;
using Makc2022.Layer3.Sql.Sample.Entities.DummyManyToMany;
using Makc2022.Layer3.Sql.Sample.Entities.DummyOneToMany;

namespace Makc2022.Layer4.Sql.Domains.DummyMain.Queries.Item.Get
{
    /// <summary>
    /// Выходные данные запроса на получение элемента в домене.
    /// </summary>
    public class DomainItemGetQueryOutput
    {
        #region Properties

        /// <summary>
        /// Объект сущности "DummyMain".
        /// </summary>
        public DummyMainEntityObject? ObjectOfDummyMainEntity { get; set; }

        /// <summary>
        /// Объекты сущности "DummyManyToMany".
        /// </summary>
        public DummyManyToManyEntityObject[]? ObjectsOfDummyManyToManyEntity { get; set; }

        /// <summary>
        /// Объекты сущности "DummyMainDummyManyToMany".
        /// </summary>
        public DummyMainDummyManyToManyEntityObject[]? ObjectsOfDummyMainDummyManyToManyEntity { get; set; }

        /// <summary>
        /// Объект сущности "DummyOneToMany".
        /// </summary>
        public DummyOneToManyEntityObject? ObjectOfDummyOneToManyEntity { get; set; }

        #endregion Properties
    }
}
